src/components/nodes-table/nodes-table.js

```diff
@@ -14,40 +13,42 @@ export default {
     ColumnFilter,
     ExpandedNode
   },
-  data: () => ({
-    settings: new Settings(localStorage),
-    showHidden: undefined,
-    itemsPerPage: undefined,
-    columns: undefined,
-    groupBy: undefined,
-    expanded: [],
-    filters: {},
-    sorting: {},
-    selected: [],
-    headersMenu: false,
-    headers: [
-      { text: 'ID', type: 'number', value: 'id', groupable: false },
-      { text: 'Manufacturer', type: 'string', value: 'manufacturer' },
-      { text: 'Product', type: 'string', value: 'productDescription' },
-      { text: 'Product code', type: 'string', value: 'productLabel' },
-      { text: 'Name', type: 'string', value: 'name' },
-      { text: 'Location', type: 'string', value: 'loc' },
-      { text: 'Secure', type: 'boolean', value: 'isSecure' },
-      { text: 'Beaming', type: 'boolean', value: 'isBeaming' },
-      { text: 'Failed', type: 'boolean', value: 'failed' },
-      { text: 'Status', type: 'string', value: 'status' },
-      { text: 'Interview stage', type: 'string', value: 'interviewStage' },
-      {
-        text: 'Last Active',
-        type: 'date',
-        value: 'lastActive',
-        groupable: false
-      }
-    ]
-  }),
+  data: function () {
+    return {
+      settings: new Settings(localStorage),
+      managedNodes: new ManagedItems(
+        this.nodes || [],
+        {
+          id: { type: 'number', label: 'ID', groupable: false },
+          manufacturer: { type: 'string', label: 'Manufacturer' },
+          productDescription: { type: 'string', label: 'Product' },
+          productLabel: { type: 'string', label: 'Product code' },
+          name: { type: 'string', label: 'Name' },
+          loc: { type: 'string', label: 'Location' },
+          isSecure: { type: 'boolean', label: 'Secure' },
+          isBeaming: { type: 'boolean', label: 'Beaming' },
+          failed: { type: 'boolean', label: 'Failed' },
+          status: { type: 'string', label: 'Status' },
+          interviewStage: { type: 'string', label: 'Interview stage' },
+          lastActive: { type: 'date', label: 'Last Active', groupable: false }
+        },
```

Comment:
@robertsLando this replaces the `headers` array and allows us to describe the properties using meta data that fits our needs without continuing to misuse the `headers` array, which is then created on-the-fly using `getTableHeaders()`.
In the future, when we are using TypeScript, we may put this information to decorators of the Z2MNode class and read it from there.